code/modules/urist/modules/GOAI/integrations/base_commander/mob_commander/obstacle_handlers/generic.dm

```diff
@@ -0,0 +1,60 @@
+/datum/goai/mob_commander/proc/HandlePry(var/datum/ActionTracker/tracker, var/obj/target)
+	log_debug("Pry task for [target]")
+	if(src.brain.GetMemoryValue(MEM_OBSTRUCTION, null) == target)
+		src.brain.DropMemory(MEM_OBSTRUCTION)
+	tracker.SetDone()
+	return
+	//TODO
+
+/datum/goai/mob_commander/proc/HandleScrew(var/datum/ActionTracker/tracker, var/obj/target)
+	log_debug("Screw task for [target]")
+	tracker.SetDone()
+	return
+	//TODO
+
+/datum/goai/mob_commander/proc/HandleGenericBreak(var/datum/ActionTracker/tracker, var/obj/target, var/method)
+	log_debug("Break task for [target]")
+	var/mob/pawn = src.GetPawn()
+	if(!pawn || !istype(pawn))
+		return
+
+	if(!tracker)
+		return
+
+	if(tracker.IsOlderThan(src.ai_tick_delay * 10))
+		tracker.SetFailed()
+		if(src.brain.GetMemoryValue(MEM_OBSTRUCTION, null) == target)	//This should really be its own proc...
+			src.brain.DropMemory(MEM_OBSTRUCTION)
+		return
+
+	var/obj/machinery/M = target
+
+	if(isnull(target) || (istype(M) && (M.stat & BROKEN)))
+		if(!isnull(target))
+			if(src.brain.GetMemoryValue(MEM_OBSTRUCTION, null) == target)
+				src.brain.DropMemory(MEM_OBSTRUCTION)
+		tracker.SetDone()
+		return
+
+	//Cooldown check
+	var/last_action = tracker.BBGet("LastAttack", null)
+	if(last_action && (world.time < last_action + DEFAULT_ATTACK_COOLDOWN))
+		return
+
+	//See what's in the mob's hands
+	var/obj/item/AH = pawn.get_active_hand()
+	var/obj/item/IH = pawn.get_inactive_hand()
```

Comment:
Just curious, does that work for the generic /mob type? I would have thought only humanoids would support this!

Reply:
It actually does!
Definition is from `/mob/proc/get_inactive_hand()`

There are some cases I could see being an issue in the future though, so might it be an idea at some point to make a new `mob_commander` subtype designed to be used with humans/humanoids only, and have them run different handlers to the parent type? Either that or the handlers get expanded to typecast the pawn itself?

Reply:
I want to keep AI polymorphous w.r.t. the pawns, the pawn type shouldn't leak through to the AI. Type-casting is good enough for now.

Although now that you mention it - we could make a submodule class for the AI that gets instantiated based on the pawn class and handles the type-castey bits.

So, for example, instead of directly adding an action, we'd go through the associate that would return e.g. mob-flavored/object-flavored/humanoid-flavored, etc. variants of each Action and/or provide a unified API to do things for all different types where possible. 

Option 2 is similar but inverted - rather than creating an ACL class based on the pawn type, we inject a set of Interface objects into atoms one way or another and just use these in handlers. This is effectively the same pattern as the Cover/Blocker stuff.